### DISCUSSION TITLE
[Feature] CLI recursive upload with auto albums, subdirs should not create own albums

### DISCUSSION THREAD

**Discussion by rcodehub** [immich-app/immich#2360] [Feature Request]
### The feature

I just started playing with Immich, awesome work so far.  I found the CLI to automatically upload from existing files and it worked great.. recursive upload with automatic album creation got my photo library into Immich with ease.

The only issue that would be wonderful if there was a resolution for is that _every_ sub directory has an album created for it.  One potential solution would be to give an option for top level only directory album creation, but that would also require the capability of albums to have subfolders.

For example, here's a filesystem of photos that was CLI uploaded:

Photo Library\
  \Vacation - Destination
    \Area 1
    \Area 2
    \Area 3

In this example an album is created for Vacation - Destination, Area 1, Area 2, Area 3 when ideally the only album created would be for Vacation - Destination and the subfolders Area 1, Area 2, Area 3 all stay in tact as sub albums under Vacation - Destination

Another mostly acceptable solution would be for photos in subfolders (Area 1, Area 2, Area 3) are all added in the main album of the parent sub folder (Vacation - Destination) so at least all photos are in the same main album.  Having an album for Vacation - Destination and then another random album for Area 3 makes no sense if you're trying to view things related to Vacation - Destination.

Hope this all makes sense and can be something to work on!

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by HariSeldon85** [comment]
I’m starting to move my family photo database , I’ve hundreds of thousands of photos, divided in folder structure.

obviously many of them have some subfolder in it that mess up the album creation.

this feature is so much needed to import first time all your photos in Immich.

**Comment by akemper** [comment]
To keep things simple for the beginning, I suggest to just add an option to Immich CLI to name albums according to the entire (recursive) folder path and not just from the directory including the imported files, e. g. instead of currently

/Vacation/2026/Climbing -> Album "Climbing"

it should be

/Vacation/2026/Climbing -> Album "Vacation 2026 Climbing" or similar.

The required changes to my understanding only affect the Immich CLI itself, not the core / DB or any type of frontend.

**Reply by NathanSweet** [reply]
You can already do this using `--folder-as-album=PATH --album-path-joiner=" "`. However, the problem I found with that is if I run it in the root folder by specifying `.` then I get names `. Vacation 2026 Climbing`. It shouldn't include the dot!

**Reply by akemper** [reply]
Interesting to hear. Which version are you referring to? Because at least in the main branch I can't even see the options `--album-path-joiner` or `--folder-as-tags` ? Instead, I already started playing a bit with my local copy of /cli/src/commands/assets.ts to find a suitable hack for my existing folder structure.

**Reply by NathanSweet** [reply]
Oh sorry, I was thinking of `immich-go`, which has those commands. Eg:
```
immich-go upload from-folder --folder-as-album=PATH --album-path-joiner=" " --server=http://10.0.0.10:2283 --api-key=yourkey --dry-run .
```
Review log file and remove `--dry-run` when confident it'll do what you want.

**Reply by akemper** [reply]
Ah ok. Thanks for pointing me there. immich-go seems to be the better starting point for me.

**Comment by NathanSweet** [comment]
I also need this. Only my first level folder is the album, other subfolders are not -- I'd like those as tags.

As a workaround I can try importing with `--folder-as-tags` and then manually create each album, but this is pretty tedious. Maybe it can be scripted.

**Reply by akemper** [reply]
So far I didn't even think about (auto) generating tags, but as you mention this it could be interesting to do this from included folders, e. g. /Party/Birthdays/2024/Jan -> #Party, #Birthdays, #2024, #Jan

**Reply by NathanSweet** [reply]
I did it using `--folder-as-tags` in `immich-go`. Afterward I went to manually create the albums and... where are the tags?! There is no UI for tag search? It may need to be a search. How I wish I had used `--folder-as-album` then deleted the albums for subfolders. I wonder if I can run `immich-go` again, it should skip uploading existing files and just create albums.
